x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/stream_enrichment_state_machine/stream_enrichment_state_machine.ts

```diff
@@ -348,7 +457,21 @@ export const streamEnrichmentMachine = setup({
               initial: 'viewDataPreview',
               on: {
                 'simulation.refresh': {
-                  actions: [{ type: 'refreshDataSources' }],
+                  actions: [
+                    { type: 'refreshDataSources' },
+                    enqueueActions(({ enqueue }) => {
+                      enqueue({ type: 'applyFieldRestorationFromPersistentMappings' });
```

Comment:
I'm a bit worried we need to call this from so many places. What about moving the persisted field mappings (small nit: I would call them "previousMappings" or so) into the simulation machine? That one seems to know better when to look at previous mappings.

The way it's wired up right now is a bit messy - it feels like the simulation state machine really wants to own this information and we need to draw a lot of wires to keep it in the enrichment state machine.